search/src/main/scala/weco/api/search/elasticsearch/ImagesQuery.scala

```diff
@@ -49,23 +48,32 @@ case object ImageSimilarity {
       )
   }
 
-  def features: (String, IndexedImage, Index) => Query =
+  def features: (String, IndexedImage, Index) => JsonObject =
     knnQuery("query.inferredData.reducedFeatures")
 
   private def knnQuery(
     field: String
-  )(imageId: String, image: IndexedImage, index: Index): Query = {
-    val query = Json.obj(
-      "knn" -> Json.obj(
-        "field" -> field.asJson,
-        "query_vector" -> image.reducedFeatures.asJson,
-        "k" -> 10.asJson,
-        "num_candidates" -> 100.asJson
+  )(imageId: String, image: IndexedImage, index: Index): JsonObject =
+    Json
+      .obj(
+        "knn" -> Json.obj(
+          "field" -> field.asJson,
+          "query_vector" -> image.reducedFeatures.asJson,
+          "k" -> 10.asJson,
+          "num_candidates" -> 100.asJson,
+          "filter" -> Json.obj(
+            "bool" -> Json.obj(
+              "must_not" -> Json.obj(
+                "ids" -> Json.obj(
+                  "values" -> Json.arr(Json.fromString(imageId))
+                )
+              )
+            )
+          )
```

Comment:
🥳 

The amount of `Json.method` we're having to sprinkle around is a bit icky, but that's a problem for another day.